src/Microsoft.TestPlatform.Common/Utilities/FakesUtilities.cs

```diff
@@ -0,0 +1,214 @@
+            IEnumerable<string> sources)
+        {
+            // If user provided fakes settings don't do anything
+            if (XmlRunSettingsUtilities.ContainsDataCollector(runSettings.CreateNavigator(), FakesMetadata.DataCollectorUri))
+            {
+                return false;
+            }
+
+            Func<IEnumerable<string>, string> configurator;
+
+            // fakes supported?
+            if (!TryGetFakesDataCollectorConfigurator(out configurator))
+            {
+                return false;
+            }
+
+            // if no fakes, return settings unchanged
+            var fakesConfiguration = configurator(sources);
+            if (fakesConfiguration == null)
+            {
+                return false;
+            }
+
+            // integrate fakes settings in configuration
+            // if the settings don't have any data collector settings, populate with empty settings
+            EnsureSettingsNode(runSettings, new DataCollectionRunSettings());
+
+            // embed fakes settings
+            var fakesSettings = CreateFakesDataCollectorSettings();
+            var doc = new XmlDocument();
+            using (var xmlReader = XmlReader.Create(
+                new StringReader(fakesConfiguration),
+                new XmlReaderSettings() { CloseInput = true }))
+            {
+                doc.Load(xmlReader);
+            }
+
+            fakesSettings.Configuration = doc.DocumentElement;
+            XmlRunSettingsUtilities.InsertDataCollectorsNode(runSettings.CreateNavigator(), fakesSettings);
+            return true;
+        }
+
+        /// <summary>
+        /// Ensures that an xml element corresponding to the test run settings exists in the setting document.
+        /// </summary>
+        /// <param name="settings">settings</param>
+        /// <param name="settingsNode">settingsNode</param>
+        private static void EnsureSettingsNode(XmlDocument settings, TestRunSettings settingsNode)
+        {
+            Debug.Assert(settingsNode != null, "Invalid Settings Node");
+            Debug.Assert(settings != null, "Invalid Settings");
+
+            var root = settings.DocumentElement;
+            if (root[settingsNode.Name] == null)
+            {
+                var newElement = settingsNode.ToXml();
+                XmlNode newNode = settings.ImportNode(newElement, true);
+                root.AppendChild(newNode);
+            }
+        }
+
+        private static bool TryGetFakesDataCollectorConfigurator(out Func<IEnumerable<string>, string> configurator)
+        {
+            try
+            {
+                var assemblyPath = Path.Combine(Path.GetDirectoryName(Assembly.GetEntryAssembly().Location), FakesConfiguratorAssembly + ".dll");
+
+                Assembly assembly = null;
+#if NET46
+                assembly = Assembly.LoadFrom(assemblyPath);
+#else
+                // I think we should not do this, since FakeConfigurator is built for net46?
```

Comment:
If we remove this then we can remove reference to System,Reflection, I did it for now so that if we do work for fakes in NetCore, we won't need to make change here.
Either way is fine with me.